### ISSUE TITLE
"aft_iam_roles.module.ct_management_service_role does not declare a provider named" warning during upgrading to 1.4.0 latest module

### ISSUE THREAD

**Issue by kyrylo-hulak** [aws-ia/terraform-aws-control_tower_account_factory#163]
**AFT Version:**
1.3.4
**Terraform Version & Provider Versions**

`terraform version`
```
terraform version
Terraform v1.0.11
on darwin_amd64
+ provider registry.terraform.io/hashicorp/archive v2.2.0
+ provider registry.terraform.io/hashicorp/aws v3.74.3
+ provider registry.terraform.io/hashicorp/local v2.2.2
+ provider registry.terraform.io/hashicorp/random v3.1.0
+ provider registry.terraform.io/hashicorp/time v0.7.2
```

`terraform providers`
```
Providers required by configuration:
.
├── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
└── module.aft_pipeline
    ├── provider[registry.terraform.io/hashicorp/local]
    ├── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
    ├── module.aft_iam_roles
    │   ├── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
    │   ├── module.audit_service_role
    │   │   └── provider[registry.terraform.io/hashicorp/aws]
    │   ├── module.ct_management_exec_role
    │   │   └── provider[registry.terraform.io/hashicorp/aws] >= 2.7.0
    │   ├── module.ct_management_service_role
    │   │   └── provider[registry.terraform.io/hashicorp/aws]
    │   ├── module.log_archive_exec_role
    │   │   └── provider[registry.terraform.io/hashicorp/aws] >= 2.7.0
    │   ├── module.log_archive_service_role
    │   │   └── provider[registry.terraform.io/hashicorp/aws]
    │   ├── module.aft_exec_role
    │   │   └── provider[registry.terraform.io/hashicorp/aws] >= 2.7.0
    │   ├── module.aft_service_role
    │   │   └── provider[registry.terraform.io/hashicorp/aws]
    │   └── module.audit_exec_role
    │       └── provider[registry.terraform.io/hashicorp/aws] >= 2.7.0
    ├── module.aft_account_request_framework
    │   ├── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
    │   └── provider[registry.terraform.io/hashicorp/time]
    ├── module.aft_backend
    │   └── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
    ├── module.aft_ssm_parameters
    │   └── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
    ├── module.aft_code_repositories
    │   ├── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
    │   └── provider[registry.terraform.io/hashicorp/local]
    ├── module.aft_customizations
    │   ├── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
    │   └── provider[registry.terraform.io/hashicorp/local]
    ├── module.aft_feature_options
    │   └── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
    ├── module.aft_account_provisioning_framework
    │   └── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
    ├── module.packaging
    │   └── provider[registry.terraform.io/hashicorp/archive]
    └── module.aft_lambda_layer
        ├── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
        ├── provider[registry.terraform.io/hashicorp/random]
        └── provider[registry.terraform.io/hashicorp/local]

Providers required by state:

    provider[registry.terraform.io/hashicorp/local]

    provider[registry.terraform.io/hashicorp/random]

    provider[registry.terraform.io/hashicorp/time]

    provider[registry.terraform.io/hashicorp/aws]

    provider[registry.terraform.io/hashicorp/archive]

╷
│ Warning: Provider aws is undefined
│ 
│   on .terraform/modules/aft_pipeline/modules/aft-iam-roles/iam.tf line 62, in module "ct_management_service_role":
│   62:     aws = aws.ct_management
│ 
│ Module module.aft_pipeline.module.aft_iam_roles.module.ct_management_service_role does not declare a provider named aws.
│ If you wish to specify a provider configuration for the module, add an entry for aws in the required_providers block within the module.
│ 
│ (and 3 more similar warnings elsewhere)
```

**Bug Description**
Terraform Validate is throwing the Warning which is not expected to see.

**To Reproduce**
1. Go to the main "aft_pipeline" module and set the latest release to 1.4.0 to upgrade.
2. then, "terraform init" to download the module upgrades.
3. next "terraform validate"

**Expected behavior**
terraform validate is successfull.

**Error Message**
```
Warning: Provider aws is undefined
│ 
│   on .terraform/modules/aft_pipeline/modules/aft-iam-roles/iam.tf line 62, in module "ct_management_service_role":
│   62:     aws = aws.ct_management
│ 
│ Module module.aft_pipeline.module.aft_iam_roles.module.ct_management_service_role does not declare a provider named aws.
│ If you wish to specify a provider configuration for the module, add an entry for aws in the required_providers block within the module.
│ 
│ (and 3 more similar warnings elsewhere)
```
Thank you!

**Comment by balltrev** [comment]
Thanks for calling this out @kyrylo-hulak. We're currently working to address all warnings thrown by Terraform during validate

**Comment by kyrylo-hulak** [comment]
The main question is Whether I can safely ignore above warnings and proceed with upgrading to 1.4.0 or I have to wait for the resolution. Thank you!

**Comment by snebhu3** [comment]
We just released AFT [v1.4.1](https://github.com/aws-ia/terraform-aws-control_tower_account_factory/releases/tag/1.4.1) which resolves the `Warning: Provider aws is undefined` warnings.

**Comment by kyrylo-hulak** [comment]
Awesome News! Thank you!
